### ISSUE TITLE
Add support for placeholder in command arguments

### ISSUE THREAD

**Issue by asamuzaK** [asamuzaK/withExEditorHost#71]
See [#122](https://github.com/asamuzaK/withExEditor/issues/122)

Things need to be considered

* Is placeholder used
* Support placeholder with or without brackets (`${foo}` or `$foo`)
* Is placeholder within quoted argument (need escape?)
* Does placeholder contain space(s) and/or backslash(es)
* Is Windows or not

**Comment by asamuzaK** [comment]
Try build
```
npm i -g withexeditorhost@dev
```

Set editorconfig.json like
```
{
  "editorPath": "/usr/bin/termite",
  "cmdArgs": ["--exec=\"nvim ${file}\""]
}
```

**Comment by asamuzaK** [comment]
@mb720 
Can you try with the dev build?
Thanks

**Comment by mb720** [comment]
Here's what I did:
    sudo npm i -g withexeditorhost@dev

This produced a message containing "withexeditorhost@5.3.0-a.4"
Then:
    cd /usr/lib/node_modules/withexeditorhost
    node index setup
In the installation dialog I selected Firefox and let withExEditorHost overwrite all the files.

The content of `~/.config/withexeditorhost/config/firefox/withexeditorhost.sh` is now
`/usr/bin/node /usr/lib/node_modules/withexeditorhost/index.js`

Calling `/usr/bin/node /usr/lib/node_modules/withexeditorhost/index.js` produces

    {"withexeditorhost":{"message":"getEditorConfig","status":"ready"}}

`~/.config/withexeditorhost/config/firefox/editorconfig.json` is
    {
      "editorPath": "/usr/bin/termite",
      "cmdArgs": ["--exec=\"nvim ${file}\""]
   }


To make sure that the new version of withExEditorHost is executed I renamed
`~.local/bin/withExEditorHost` to `~/.local/bin/withExEditorHostOld`.

After reloading withExEditor in Firefox and starting it by right-clicking into a text field and selecting "Edit text with termite" nothing seems to happen. The console doesn't produce output.

I tried withExEditor with the old contents of `~/.config/withexeditorhost/config/firefox/editorconfig.json` containing gVim as the editor: That worked fine.

Let me know how I can debug this further. Is there a way to display the version of withExEditorHost? Just to be sure I'm really using the dev build.

**Comment by asamuzaK** [comment]
Thanks for the detailed research.

Comment from [Text file name as placeholder · Issue #122](https://github.com/asamuzaK/withExEditor/issues/122):

> This would work:
> 
> > /usr/bin/termite --exec="nvim /tmp/withExEditor/1822667/tmpFiles/1/139/github.com/new_comment_field.txt"

Can you check that the dev build is spawning command like that?

> Let me know how I can debug this further. 

Unfortunately, on WebExtensions, there is no simple way to debug a host using Firefox.

Anyway, you are running dev build.
You don't have to worry about that.

Here is the way how Firefox communicates with the host:

Firefox looks for the host manifest which is stored in a particular place.
On Linux it is:
`~/.mozilla/native-messaging-hosts/withexeditorhost.json`.

Open that file and check the `path` field.
It should point to the shell script which is created by host on install:
```
"path": "~/.config/withexeditorhost/config/firefox/withexeditorhost.sh",
```

**Comment by asamuzaK** [comment]
Another try build, 5.3.0-a.5

Added option which runs command within shell or not.
Run setup, choose `Y` for `Run command inside of a shell?` question. and see what happens?

**Comment by mb720** [comment]
It's working now, great! 🥳 🎉

I found the reason why it didn't before: `${file}` is the *already quoted* file path.

The following `editorconfig.json` didn't work because I quoted the file again which caused nvim to not find the file and exit:

    {
      "editorPath": "/usr/bin/termite",
      "cmdArgs": ["--exec=\"nvim ${file}\""]
    }

If the implementation of `${file}` stays as it is, I'd say this fact should be well documented, let me know if I can help you with that.

This configuration on the other hand works fine (it's what I'm using now to write this comment):

    {
      "editorPath": "/usr/bin/termite",
      "cmdArgs": ["--exec=nvim ${file}"]
    }

I also added `"shell": true` but I didn't notice a difference in behavior.

This works great too (but being a GUI program, gVim already worked before):

    {
      "editorPath": "/usr/bin/gvim",
      "cmdArgs": ["${file}"]
    }


Thanks for adding this feature! 👍

---

I'm using withexeditorhost@5.3.0-a.5.
